src/bokeh/io/notebook.py

```diff
@@ -498,7 +501,10 @@ def publish_display_data(data: dict[str, Any], metadata: dict[Any, Any] | None =
     from IPython.display import publish_display_data
     publish_display_data(data, metadata, transient=transient, **kwargs)
 
-def show_app(app: Application, state: State, notebook_url: str | Callable[[int | None], str], port: int = 0, **kw: Any) -> None:
+
+ProxyUrlFunc: TypeAlias = Callable[[int | None], str]
+
+def show_app(app: Application, state: State, notebook_url: [str | ProxyUrlFunc] = "localhost:8888", port: int = 0, **kw: Any) -> None:
```

Comment:
No braces 
```suggestion
def show_app(app: Application, state: State, notebook_url: str | ProxyUrlFunc = "localhost:8888", port: int = 0, **kw: Any) -> None:
```

Reply:
Also we have a fairly long line length limit in order to accommodate some things, but this line is now long enough that this might be prefered
```python
def show_app(
    app: Application, 
    state: State, 
    notebook_url: [str | ProxyUrlFunc] = "localhost:8888", 
    port: int = 0, 
    **kw: Any
) -> None:
```